### PR TITLE
BUG: Fix tripleg generation

### LINKED ISSUE #280: BUG: index selection in generate_triplegs
generate_triplegs occasionally throws the error `LineStrings must have at least 2 coordinate tuples` even though we filtered out the invalid triplegs in [line 363](https://github.com/mie-lab/trackintel/blob/8f53d0cc6495e53df06554a76adbd66a57dd8e2d/trackintel/preprocessing/positionfixes.py#L363).

This is because in all the `cond_*` functions we deal with records based on their index in the array but not with the df.index values. However, In [line 365](https://github.com/mie-lab/trackintel/blob/8f53d0cc6495e53df06554a76adbd66a57dd8e2d/trackintel/preprocessing/positionfixes.py#L365), we regard `cond_to_remove` as `pfs.index` values, and this will delete the wrong pfs if the `pfs.index` is not continues or not starting from 0. 

The proposed fix would be to replace the `pfs.loc[pfs.index.isin(cond_to_remove), "tripleg_id"] = -1` into `pfs.loc[pfs.index[cond_to_remove], "tripleg_id"] = -1` in Line 365.

A new test for this will follow.

### PR DESCRIPTION
**Pull request by hong2223**
closes #280

### CODE REVIEW

File: tests/preprocessing/test_positionfixes.py [hunk 1]
```diff
@@ -45,8 +45,14 @@ def example_positionfixes():
 
 
 @pytest.fixture
-def isolated_positionfixes():
-    """Positionfixes with two isolated positionfixes that have the same geometry but different timestamps."""
+def example_positionfixes_isolated():
+    """
+    Positionfixes with isolated positionfixes.
+
+    User1 have the same geometry but different timestamps.
```
Comment (henrymartin1) [inline tests/preprocessing/test_positionfixes.py:52]: have -> has

File: tests/preprocessing/test_positionfixes.py [hunk 2]
```diff
@@ -45,8 +45,14 @@ def example_positionfixes():
...
+    User2 have different geometry and timestamps.
+    User3 have only one isolated positionfixes.
```
Comment (henrymartin1) [inline tests/preprocessing/test_positionfixes.py:54]: positionfix singular if it is only 1

File: tests/preprocessing/test_positionfixes.py [hunk 3]
```diff
@@ -45,8 +45,14 @@ def example_positionfixes():
...
+    """
```
Comment (henrymartin1) [inline tests/preprocessing/test_positionfixes.py:55]: p3 and p4 below are the same. If this is intentional I would rather only use p3 below

File: tests/preprocessing/test_positionfixes.py [hunk 4]
```diff
@@ -228,9 +237,31 @@ def test_temporal(self):
...
 
-    def test_invalid_isolates(self, isolated_positionfixes):
+    def test_noncontinuous_unordered_index(self, example_positionfixes_isolated):
+        """The unordered and noncontinuous index of pfs shall not affect the generate_triplegs() result."""
+        pfs = example_positionfixes_isolated
+
+        # regenerate noncontinuous index
+        pfs.index = range(0, pfs.shape[0] * 2, 2)
+        pfs.index.name = "id"
+        # unorder index
```
Comment (henrymartin1) [inline tests/preprocessing/test_positionfixes.py:247]: I would say "shuffle"

File: tests/preprocessing/test_positionfixes.py [hunk 5]
```diff
@@ -228,9 +237,31 @@ def test_temporal(self):
 class TestGenerate_triplegs:
     """Tests for generate_triplegs() method."""
 
-    def test_invalid_isolates(self, isolated_positionfixes):
+    def test_noncontinuous_unordered_index(self, example_positionfixes_isolated):
+        """The unordered and noncontinuous index of pfs shall not affect the generate_triplegs() result."""
```
Comment (henrymartin1) [inline tests/preprocessing/test_positionfixes.py:241]: I think it is "unordered" but "non-continuous"

File: tests/preprocessing/test_positionfixes.py [hunk 6]
```diff
@@ -228,9 +237,31 @@ def test_temporal(self):
...
+        warn_string = "The positionfixes with ids .* lead to invalid tripleg geometries."
+        with pytest.warns(UserWarning, match=warn_string):
+            pfs, tpls = pfs.as_positionfixes.generate_triplegs()
+
+        # the index shall be reordered
+        assert np.all(pfs.index == range(0, pfs.shape[0] * 2, 2))
+
+        # only user 1 has generated a tripleg.
+        # user 0 and 2 tripleg has invalid geometry (two identical points and only one point respectively)
+        assert (tpls["user_id"].unique()[0] == 1) and (len(tpls) == 1)
```
Comment (henrymartin1) [inline tests/preprocessing/test_positionfixes.py:260]: I think `(tpls["user_id"].unique()[0] == 1)` is quite confusing. How about `(tpls["user_id"].iloc[0] == 1)`

**Review by henrymartin1 — CHANGES_REQUESTED** [review]
Looks good to me! I only have minor spelling stuff

File: tests/preprocessing/test_positionfixes.py (hunk 5 above)
Comment (hong2223) [reply tests/preprocessing/test_positionfixes.py:240]: I am not an expert in English, but in the dictionary it says [noncontinuous](https://www.collinsdictionary.com/dictionary/english/noncontinuous)

File: tests/preprocessing/test_positionfixes.py (hunk 3 above)
Comment (hong2223) [reply tests/preprocessing/test_positionfixes.py:55]: strange legacy issue. Already changed

File: tests/preprocessing/test_positionfixes.py (hunk 6 above)
Comment (hong2223) [reply tests/preprocessing/test_positionfixes.py]: the former implies there is only one `user_id` whereas the latter only takes the first record. It is true they are the same under the condition `(len(tpls) == 1)`.

How about `(tpls["user_id"].unique() == [1])`?

Comment (henrymartin1) [reply tests/preprocessing/test_positionfixes.py]: `tpls["user_id"].unique() == [1]` works for me.

**Review by henrymartin1 — APPROVED** [review]
Thanks @hong2223 for the quick fix!

**Pull request merged**